elife/gearman-client-php.sls

```diff
@@ -39,9 +44,51 @@ php-ppa-gearman:
             - test -e /etc/apt/sources.list.d/ondrej-ubuntu-pkg-gearman-xenial.list
         - require_in:
             - pkg: gearman-php-extension
-{% endif %}
 
 gearman-php-extension:
     pkg.installed:
         - pkgs:
             - php-gearman
+
+{% elif salt['grains.get']('osrelease') == '18.04' %}
+
+php-ppa-gearman:
+    pkg.purged:
+        - pkgs:
+            - php-gearman
+        - require_in:
+            - gearman-php-extension
+
+gearman-php-extension:
+    git.latest:
+        - name: https://github.com/wcgallego/pecl-gearman
+        - rev: gearman-2.0.5
+        - branch: master
+        - target: /opt/pecl-gearman
+        - force_fetch: true
+        - force_clone: true
+        - force_reset: true
+        - force_checkout: true
+    pkg.installed:
+        - pkgs:
+            - libgearman-dev
+    file.managed:
+        - name: /etc/php/7.2/mods-available/gearman.ini
+        - contents: extension=gearman.so
+        - require:
+            - php
+    cmd.run:
+        - cwd: /opt/pecl-gearman
+        - name: |
```

Comment:
this becomes a bash script, so I think you need a `set -e` as the first line to make errors interrupt it

Reply:
done